libevmasm/Assembly.cpp

```diff
@@ -817,7 +830,9 @@ std::map<u256, u256> const& Assembly::optimiseInternal(
 				else
 					copy(orig, iter, back_inserter(optimisedItems));
 			}
-			if (optimisedItems.size() < m_items.size())
+			auto optimisedItemsCost = static_cast<bigint>(optimisedItems.size()) + runs * runGas(optimisedItems, KnownState{}).value;
+			auto itemsCost = static_cast<bigint>(m_items.size()) + runs * runGas(m_items, KnownState{}).value;
```

Comment:
If we want to try and be accurate about this, the code size component of the cost should be calculated similarly to https://github.com/ethereum/solidity/blob/272892e98e3f0f88be8aca063aad84008eadfad3/libevmasm/Inliner.cpp#L178, i.e. based on an estimate of the actual number of bytes required for each assembly item, times the actual cost of a byte of code...